### PR TITLE
Integrates "Intercom" New AI Messenger (Customer Support Chat System)

### PR DESCRIPTION
**Pull request by coverlandjohnlee**
Notes:
-adds new npm package
-we are using the "visitors" integration (not have to be logged in to use the chat)
-Isaac has already approved the messenger itself
-I kept the old messenger files with comments at the top, in case we wanted to keep them around. Can remove the files entirely if wanted

https://app.clickup.com/t/8688yrn0v

For Testing:
Please make sure that the frontend is just useable on both desktop and mobile
It's the blue messenger icon at bottom right

### CODE REVIEW

File: src/components/LiveChat.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// old messenger
 'use client';
 import { useLiveChatContext } from '@/contexts/LiveChatContext';
 import dynamic from 'next/dynamic';
```
Comment (george-icarcover) [inline src/components/LiveChat.tsx:4]: I don't think you actually have to keep this part, unless they said they might want to change back in the future

File: src/contexts/LiveChatContext.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// old messenger
 'use client';
 import React, {
   Dispatch,
```
Comment (george-icarcover) [inline src/contexts/LiveChatContext.tsx:4]: May want to remove

**Review by george-icarcover — APPROVED** [review]
LGTM, may want to remove the old code though

**Comment by coverlandjohnlee** [comment]
> LGTM, may want to remove the old code though

cool thanks merging

**Pull request merged**